src/main/java/seedu/duke/commands/EditCommand.java

```diff
@@ -0,0 +1,120 @@
+import seedu.duke.exceptions.NoSuchModuleException;
+import seedu.duke.tasks.Module;
+import seedu.duke.tasks.ModuleList;
+import seedu.duke.tasks.Task;
+import seedu.duke.tasks.TaskList;
+import seedu.duke.util.StringConstants;
+
+import java.util.Objects;
+
+public class EditCommand extends Command {
+
+    private static final String EDIT_MODULE_SUCCESS = StringConstants.EDIT_MODULE_SUCCESS;
+    private static final String EDIT_TASK_SUCCESS = StringConstants.EDIT_TASK_SUCCESS;
+    private static final String EDIT_TASK_WITH_MODULE_SUCCESS = StringConstants.EDIT_TASK_WITH_MODULE_SUCCESS;
+    private static final String TASK_DESCRIPTION = StringConstants.TASK_DESCRIPTION_STR;
+    private static final String ESTIMATED_WORKING_TIME = StringConstants.ESTIMATED_WORKING_TIME_STR;
+    private static final String TASK_NAME = StringConstants.TASK_NAME_STR;
+
+    private String moduleCode;
+    private String taskModule;
+    private int taskIndex = -1;
+    private String taskParameter;
+    private String result = "";
+    private boolean isGeneralTask = false;
+    private final String changedParameter;
+
+    public int getTaskIndex() {
+        return taskIndex;
+    }
+
+    public String getModuleCode() {
+        return moduleCode;
+    }
+
+    public String getTaskModule() {
+        return taskModule;
+    }
+
+    public EditCommand(String moduleCode, String description) {
+        this.moduleCode = moduleCode;
+        this.changedParameter = description;
+    }
+
+    public EditCommand(String taskModule, int taskIndex, String description, String workingTime, String taskName) {
+        this.taskModule = taskModule;
+        this.taskIndex = taskIndex;
+        if (!Objects.isNull(description)) {
+            this.taskParameter = TASK_DESCRIPTION;
+            this.changedParameter = description;
+            assert Objects.isNull(workingTime);
+            assert Objects.isNull(taskName);
+        } else if (!Objects.isNull(workingTime)) {
+            this.taskParameter = ESTIMATED_WORKING_TIME;
+            this.changedParameter = workingTime;
+            assert Objects.isNull(taskName);
+        } else {
+            this.taskParameter = TASK_NAME;
+            this.changedParameter = taskName;
+        }
+    }
+
+    @Override
+    public CommandResult execute(ModuleList moduleList) throws ModHappyException {
+        if (taskIndex < 0) {
+            editModuleDescription(moduleList);
+        } else {
+            Module targetModule;
+            if (Objects.isNull(taskModule)) {
+                targetModule = moduleList.getGeneralTasks();
+                isGeneralTask = true;
+            } else {
+                targetModule = moduleList.getModule(taskModule);
+                if (Objects.isNull(targetModule)) {
+                    throw new NoSuchModuleException();
+                }
+            }
+            editTaskFromModule(targetModule);
+        }
+        return new CommandResult(result);
```

Comment:
Maybe using booleans would be better to reduce nesting. Here's a quick modification I did:
[EditCommand.zip](https://github.com/AY2122S2-CS2113T-T10-3/tp/files/8255488/EditCommand.zip)
Not sure if this would be better, as it will make the execute implementation look different from the other commands' execute implementation.